File: single-language-analysis/dataset-files/python/test_smell_016.py
import pytest
import threading
import time
import logging

logging.basicConfig(level=logging.INFO)
log = logging.getLogger()

stop = False

class TestMainTest:
    def test_smell_016(self):
        global stop
        i = 0
        
        def run():
            nonlocal i
            while not stop:
                i += 1
                log.info("loop thread %d", i)
                time.sleep(0.7)
            log.info("stopThread[Thread:%s]: count i is %d", threading.current_thread().name, i)

        t = threading.Thread(target=run)
        t.start()
        
        time.sleep(7)
        stop = True
        log.info("stopThread[Thread:%s]: stopped!", t.name)

File: single-language-analysis/dataset-files/java/TestSmell016.java
// Original URL: https://github.com/zevichen/base/blob/5a7e50aab41c5949b91fd96a175cfdcd85812608/src/main/java/com/czw/base/thread/MainTest.java#L29-L52

public class TestSmell016 {
	@Test
	public void stopThread() throws InterruptedException{
		Thread t = new Thread(new Runnable() {
			public void run() {
				int i = 0;
				while(!stop){
					i++;
					log.info("loop thread {}",i);
					try {
						Thread.sleep(700);
					} catch (InterruptedException e) {
						e.printStackTrace();
					}
				}
				log.info("stopThread[Thread:{}]: count i is {}",Thread.currentThread().getName(),i);
			}
		});
		t.start();
		
		Thread.sleep(7000);
		stop = true;
		log.info("stopThread[Thread:{}]: stopped!",t.getName());
	
	}
}
